src/app/storage.js

```diff
@@ -0,0 +1,97 @@
+var utils = require('./utils');
+
+function Storage (updateFiles) {
+
+  var EDITOR_SIZE_CACHE_KEY = 'editor-size-cache';
+
+  this.rename = function (originalName, newName) {
+    var content = this.get(originalName);
+    this.set(newName, content);
+    this.remove(originalName);
+  };
+
+  this.remove = function (name) {
+    window.localStorage.removeItem(name);
+  };
+
+  this.setEditorSize = function (size) {
+    this.set(EDITOR_SIZE_CACHE_KEY, size);
+  };
+
+  this.getEditorSize = function () {
+    return this.get(EDITOR_SIZE_CACHE_KEY);
+  };
+
+  this.getFileContent = function (key) {
+    return this.get(utils.fileKey(key));
+  };
+
+  this.exists = function (key) {
+    return !!this.get(key);
+  };
+
+  this.set = function (key, content) {
+    window.localStorage.setItem(key, content);
+  };
+
+  this.get = function (key) {
+    return window.localStorage.getItem(key);
+  };
+
+  this.keys = function () {
+    return Object.keys(window.localStorage);
+  };
+
+  this.loadFile = function (filename, content) {
+    if (this.exists(filename) && this.get(filename) !== content) {
+      var count = '';
+      var otherKey = filename + count;
+      while (this.exists(filename + count)) count = count - 1;
+      this.rename(filename, filename + count);
+    }
+    this.set(filename, content);
+  };
+
+  this.sync = function () {
```

Comment:
I haven't dug into what this is actually doing. I've just been trying to preserve behavior. I agree that its intent should be clarified.